### PR TITLE
Check NVIDIA device files and modules

### PR DESCRIPTION
**Pull request by b10n1k**
Due to the not easy way to provide and install nvidia-compute-utils-G06 which provides the nvidia-smi, we check device files and the loaded modules on the system. Seems there is no better solution for the moment.

- Related ticket: https://progress.opensuse.org/issues/131501
- Verification run: https://openqa.suse.de/tests/11725797 
https://openqa.suse.de/tests/11725798

### CODE REVIEW

File: tests/publiccloud/nvidia.pm [hunk 1]
```diff
@@ -20,11 +20,15 @@ sub run {
         zypper_call('--gpg-auto-import-keys addrepo -p 90 ' . get_required_var('NVIDIA_REPO') . ' nvidia_repo');
         zypper_call '--gpg-auto-import-keys ref';
     }
-    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gsp-G06 ", quiet => 1);
+    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gspx-G06 ", quiet => 1);
     $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
 
     validate_script_output("hwinfo --gfxcard", sub { /nVidia.*Tesla T4/mg });    # depends on terraform setup
-    assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
+        # assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
```
Comment (asmorodskyi) [inline tests/publiccloud/nvidia.pm:27]: can you delete this line please ?

**Review by asmorodskyi — APPROVED** [review]
genrally LGTM , just don't like to merge commented code

File: tests/publiccloud/nvidia.pm [hunk 2]
```diff
@@ -20,11 +20,15 @@ sub run {
...
+        # nvidia-smi is delivered by nvidia-compute-utils-G06. Not available on PublicCloud
+        # Check device files and loaded modules instead.
+    assert_script_run("ls -la /dev/{nvidia-modeset,nvidia-uvm-tools,nvidiactl,nvidia-uvm,nvidia0}");
+    assert_script_run("lsmod | grep nvidia");
```
Comment (grisu48) [inline tests/publiccloud/nvidia.pm:31]: I would do this before checking the device files. If the kernel module is not loaded then the device files are not there. Also WDYT about

```suggestion
    validate_script_output("lsmod", sub { m/nvidia/ }, fail_message => "nvidia module not loaded");
```

**Review by grisu48 — COMMENTED** [review]
Two suggestions

File: tests/publiccloud/nvidia.pm (hunk 1 above)
Comment (b10n1k) [reply tests/publiccloud/nvidia.pm]: *ready for review*

File: tests/publiccloud/nvidia.pm [hunk 3]
```diff
@@ -20,11 +20,15 @@ sub run {
...
-    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gsp-G06 ", quiet => 1);
+    zypper_call("in nvidia-open-driver-G06-signed-kmp-default kernel-firmware-nvidia-gspx-G06 ", quiet => 1);
     $args->{my_instance}->softreboot(timeout => get_var('PUBLIC_CLOUD_REBOOT_TIMEOUT', 600));
 
     validate_script_output("hwinfo --gfxcard", sub { /nVidia.*Tesla T4/mg });    # depends on terraform setup
-    assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
+        # assert_script_run("LD_LIBRARY_PATH=/usr/lib/kernel-firmware-nvidia-gsp-G06 /usr/lib/kernel-firmware-nvidia-gsp-G06/nvidia-smi --query");
+        # nvidia-smi is delivered by nvidia-compute-utils-G06. Not available on PublicCloud
+        # Check device files and loaded modules instead.
+    assert_script_run("ls -la /dev/{nvidia-modeset,nvidia-uvm-tools,nvidiactl,nvidia-uvm,nvidia0}");
```
[the remark this answers is not in the document]
Comment (b10n1k) [reply tests/publiccloud/nvidia.pm]: *ready for review*

File: tests/publiccloud/nvidia.pm (hunk 2 above)
Comment (b10n1k) [reply tests/publiccloud/nvidia.pm]: *ready for review*

Comment (grisu48) [reply tests/publiccloud/nvidia.pm]: Please move this check up. If the module is not loaded, the device files are not present, so this is basically dead code. If you move it up, it acts as a preliminary check if the kernel modules is loaded before the device files are checked.

**Review by asmorodskyi — APPROVED** [review]
LGTM

File: tests/publiccloud/nvidia.pm (hunk 2 above)
Comment (b10n1k) [reply tests/publiccloud/nvidia.pm]: @grisu48 please check again. change was pushed

Comment (grisu48) [reply tests/publiccloud/nvidia.pm]: yep, looks good now. Thanks!

**Comment by asmorodskyi** [comment]
please tidy up your PR , CI pipeline is failing

**Pull request merged**